pkg/asset/installconfig/gcp/validation.go

```diff
@@ -141,66 +141,144 @@ func validateZoneProjects(client API, ic *types.InstallConfig, fieldPath *field.
 	return allErrs
 }
 
-// validateManagedZones will validate the public and private managed zones if they exist.
-func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
-	commonZoneLookup := func(dnsZone, zoneProject, defaultProject string) error {
-		project := zoneProject
-		if project == "" {
-			project = defaultProject
-		}
+// internalFindProject finds the correct project to use during installation. If the project id is
+// provided in the zone use the project id, otherwise use the default project.
+func internalFindProject(zone *gcp.DNSZone, defaultProject string) string {
+	if zone != nil && zone.ProjectID != "" {
+		return zone.ProjectID
+	}
+	return defaultProject
+}
 
-		if _, err := client.GetDNSZoneByName(context.TODO(), project, dnsZone); err != nil {
-			return err
-		}
-		return nil
+// internalFindDNSZone finds a zone in a project. If a project is provided in the zone, the project
+// is used otherwise the default project is used.
+func internalFindDNSZone(client API, zone *gcp.DNSZone, defaultProject string) (*dns.ManagedZone, string, error) {
+	if zone == nil {
+		return nil, defaultProject, fmt.Errorf("zone is not set")
 	}
```

Comment:
This check should not be necessary, the calling code is already ensuring zone != nil